File: java01t/src/step05/Test08.java
package step05;

/* 메서드 호출과 스택 
 * - 메서드 호출 시 스택에 쌓인다는 개념을 좀 더 깊이 실험.
 * - 스택 메모리 그림을 그리고 변수의 생성과 해제 상황을 묘사하라!
 */
public class Test08 {

  static void m1(int a) {
    a += 10;
    m2(a);
    System.out.printf("m1(): %d\n", a);
  }
  
  static void m2(int a) {
    a += 10;
    m3(a);
    System.out.printf("m2(): %d\n", a);
  }
  
  static void m3(int a) {
    a += 10;
    System.out.printf("m3(): %d\n", a);
  }
  
  public static void main(String[] args) {
    int a = 7;
    m1(a);

  }

}

File: node1t/src/ex1/test08.js
var http = require('http');
var url = require('url');

// 실습 목표: 이름을 파라미터로 받아서 인사하기
//
http.createServer(function(req, res) {
	console.log("------------------------------");
	var urlInfo = url.parse(req.url, true);
	var path = urlInfo.pathname;
	var name = urlInfo.query.name;
	
    res.writeHead(200, {
    		'Content-Type': 'text/html;charset=UTF-8',
    	});
    res.write('<html><head><title>test08</title></head>\n');
    res.write('<body><h1>test08</h1>\n');
    res.write('<p>안녕하세요! ' + name + '님</p>\n');
    res.end('</body></html>\n');
    
}).listen(1337, '127.0.0.1');
console.log('Server running at http://127.0.0.1:1337/');










